client/tx/factory.go

```diff
@@ -73,6 +77,9 @@ func NewFactoryCLI(clientCtx client.Context, flagSet *pflag.FlagSet) Factory {
 	feesStr, _ := flagSet.GetString(flags.FlagFees)
 	f = f.WithFees(feesStr)
 
+	tipsStr, _ := flagSet.GetString(flags.FlagTip)
+	f = f.WithTips(tipsStr, "")
```

Comment:
we are handling it here, [ref](https://github.com/cosmos/cosmos-sdk/blob/99ae8232319ffb7a703809935df98dc9f7e3129d/client/tx/tx.go#L341-L343)

Reply:
We actually do need a `--tipper` flag, because the tipper needs to use his address on the target chain. See https://github.com/cosmos/cosmos-sdk/pull/10311#discussion_r760104988

Reply:
Do we need to take it as mandatory flag or optional one?

Reply:
Tips are always optional. But if `--tip` is set, then `--tipper` must be set too, and vice versa.

Reply:
When the `--tipper` flag didn't set, should I use the `fromAddress`? (from address will have different prefix right in this case)

Reply:
> from address will have different prefix

Yes exactly, this won't work in general. I think we need to make `--tipper` be a user input (if `--tip` is set too)